### ISSUE TITLE
SolutionCrawlerService should provide a way to add an IncrementalAnalyzer in a deferred manner.

### ISSUE THREAD

**Issue by tannergooding** [dotnet/roslyn#13722]
Currently the `SolutionCrawlerService` causes all exported `IIncrementalAnalyzerProvider`s to be loaded after the solution has finished loading.

This will cause an assembly that exports an `IIncrementalAnalyzerProvider` to load, even if it is not yet needed.

The `SolutionCrawlerService` should be updated to provide an API such that an package can import the `ISolutionCrawlerService` and add a new `IIncrementalAnalyzer` when the feature is enabled and remove the `IIncrementalAnalyzer` when the feature is disabled.

The `SolutionCrawlerService` should ensure that `NewSolutionSnapshot` is called when a new `IIncrementalAnalyzer` is added.

It is understood that any `IIncrementalAnalyzer` added in this manner will not receive any prior information about the state of the `Workspace` (just the current state of the `Workspace` and the future changes sent out).

**Comment by tannergooding** [comment]
FYI. @dotnet/testimpact, @dotnet/roslyn-analysis

**Comment by ManishJayaswal** [comment]
@heejaechang @srivatsn we would need this addressed before we can do the dynamic analysis VS setup authoring so that we do not regress RPS. Can you guys prioritize this work?

**Comment by heejaechang** [comment]
I am told that this is for RC? I will make this change soon as I am done with preview 5 work.

**Comment by ManishJayaswal** [comment]
yes early in RC ( in about 2 weeks) will work.

**Comment by srivatsn** [comment]
This is done right?

**Comment by heejaechang** [comment]
this is already fixed
